lib/pg_eventstore/commands/append.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module PgEventstore
+  module Commands
+    class Append < AbstractCommand
+      # @param stream [PgEventstore::Stream]
+      # @param events [Array<PgEventstore::Event>]
+      # @param options [Hash]
+      # @option options [Integer] :expected_revision provide your own revision number
+      # @option options [Symbol] :expected_revision provide one of next values: :any, :no_stream or :stream_exists
+      # @return [Array<PgEventstore::Event>] persisted events
+      # @raise [PgEventstore::WrongExpectedRevisionError]
+      def call(stream, *events, options: {})
+        raise SystemStreamError, stream if stream.system?
+
+        queries.transaction do
+          stream = queries.find_or_create_stream(stream)
+          revision = stream.stream_revision
+          assert_expected_revision!(revision, options[:expected_revision]) if options[:expected_revision]
+          events.map.with_index do |event, index|
+            queries.insert(stream, prepared_event(event, revision + index + 1))
+          end.tap do
+            queries.update_stream_revision(stream, revision + events.size)
```

Comment:
> Why do you need this update? It is not returning the query so I can't get this value and you are not rechecking it later.

Stream's value is read above - `revision = stream.stream_revision`

> Why you are not updating it right after queries.insert? like queries.update_stream_revision(stream, revision + index)

Because it will produce `UPDATE` request on each `INSERT` instead of one `UPDATE` after all events are inserted.